pkgs/development/python-modules/k5test/fix-paths.patch

```diff
@@ -26,97 +26,100 @@ index e289fac..e2f486a 100644
-+            ('kadmind', 'kadmind', '@krb5Full@/bin/kadmind'),
-+            ('kprop', 'kprop', '@krb5Full@/bin/kprop'),
-+            ('_kinit', 'kinit', '@krb5Full@/bin/kinit'),
-+            ('_klist', 'klist', '@krb5Full@/bin/klist'),
+             ("kdb5_util", "kdb5_util", "/usr/sbin/kdb5_util"),
+             ("krb5kdc", "krb5kdc", "/usr/sbin/krb5kdc"),
+-            ("kadmin", "kadmin", "/usr/bin/kadmin"),
++            ("kadmin", "kadmin", "@krb5FullDev@/bin/kadmin"),
+             ("kadmin_local", "kadmin.local", "/usr/sbin/kadmin.local"),
+             ("kadmind", "kadmind", "/usr/sbin/kadmind"),
+             ("kprop", "kprop", "/usr/sbin/kprop"),
+-            ("_kinit", "kinit", "/usr/bin/kinit"),
+-            ("_klist", "klist", "/usr/bin/klist"),
++            ("_kinit", "kinit", "@krb5FullDev@/bin/kinit"),
++            ("_klist", "klist", "@krb5FullDev@/bin/klist"),
          ]
  
      @property
-@@ -585,12 +585,12 @@ class HeimdalRealm(K5Realm):
+@@ -628,12 +628,12 @@ class HeimdalRealm(K5Realm):
  
          return [
-             ('krb5kdc', 'kdc', os.path.join(base, 'kdc')),
--            ('kadmin', 'kadmin', '/usr/bin/kadmin'),
--            ('kadmin_local', 'kadmin', '/usr/bin/kadmin'),
-+            ('kadmin', 'kadmin', '@krb5Full@/bin/kadmin'),
-+            ('kadmin_local', 'kadmin', '@krb5Full@/bin/kadmin'),
-             ('kadmind', 'kadmind', os.path.join(base, 'kadmind')),
--            ('_kinit', 'kinit', '/usr/bin/kinit'),
--            ('_klist', 'klist', '/usr/bin/klist'),
--            ('_ktutil', 'ktutil', '/usr/bin/ktutil'),
-+            ('_kinit', 'kinit', '@krb5Full@/bin/kinit'),
-+            ('_klist', 'klist', '@krb5Full@/bin/klist'),
-+            ('_ktutil', 'ktutil', '@krb5Full@/bin/ktutil'),
+             ("krb5kdc", "kdc", os.path.join(base, "kdc")),
+-            ("kadmin", "kadmin", "/usr/bin/kadmin"),
+-            ("kadmin_local", "kadmin", "/usr/bin/kadmin"),
++            ("kadmin", "kadmin", "@krb5FullDev@/bin/kadmin"),
++            ("kadmin_local", "kadmin", "@krb5FullDev@/bin/kadmin"),
+             ("kadmind", "kadmind", os.path.join(base, "kadmind")),
+-            ("_kinit", "kinit", "/usr/bin/kinit"),
+-            ("_klist", "klist", "/usr/bin/klist"),
+-            ("_ktutil", "ktutil", "/usr/bin/ktutil"),
++            ("_kinit", "kinit", "@krb5FullDev@/bin/kinit"),
++            ("_klist", "klist", "@krb5FullDev@/bin/klist"),
++            ("_ktutil", "ktutil", "@krb5FullDev@/bin/ktutil"),
          ]
  
      @property
 diff --git a/k5test/unit.py b/k5test/unit.py
-index 3c64b9e..59da1ab 100644
+index caa8315..5bcf0bf 100644
 --- a/k5test/unit.py
 +++ b/k5test/unit.py
-@@ -38,7 +38,7 @@ _KRB_VERSION = None
+@@ -39,7 +39,7 @@ _KRB_VERSION = None
  def krb_minversion_test(target_version, problem, provider=None):
      global _KRB_VERSION
      if _KRB_VERSION is None:
 -        _KRB_VERSION = _utils.get_output("krb5-config --version")
```

Comment:
Wouldn't it work when taking it from PATH?

Reply:
This is not the test code for k5test, for that, we could set `PATH` during the build I think. It is the python module which is used by other programs for testing.
If we require that this should be used from `PATH`, we will require the other programs to put the krb5 tools (such as `krb5-config`) to their `PATH` envvar during the check phase.

Is this what you wanted?

I think k5test is opaque from the users point of view.